File: pset1/mario/C/mario_simple.c
#include <stdio.h>

int main(void) {
    
    // Height input
    int height;
    do {
        printf("Height: ");
        scanf("%d", &height);
    }
    while (height < 1 || height > 8);
    
    for (int i = 0; i < height; i ++) {

        // first row for height = 4 > "xxxy"
        // spaces = 4 - 0 - 1
        int spaces = height - i - 1;
        
        // hashes = 0 + 1
        int hashes = i + 1;

        // print spaces
        for (int j = 0; j < spaces; j ++) {
            printf(" ");
        }
        
        // print hashes
        for (int j = 0; j < hashes; j ++) {
            printf("#");
        }
        printf("\n");
    }
}

File: pset1/mario/python/mario_simple.py
# Get height
height = 0
while height < 1 or height > 8:
    height = int(input("Height:"))

for i in range(height):

    # first row if heigh = 4, "xxx#"
    # spaces = 4 - 0 - 1
    spaces = height - i - 1
    # hashes = 0 + 1
    hashes = i + 1

    # Print spaces
    for j in range(spaces):
        print(" ", end = "")
    
    # Print hashes
    for j in range(hashes):
        print("#", end = "")
    i += 1
    print("")